zypp/PathInfo.h

```diff
@@ -604,6 +604,14 @@ namespace zypp
      **/
     int rename( const Pathname & oldpath, const Pathname & newpath );
 
+    /**
+     * Like '::rename', but falls back to using mv(1) in case of EXDEV, which
+     * can occur when using OverlayFS.
+     *
+     * @return 0 on success, errno on failure
+     **/
+
+    int safe_rename( const Pathname & oldpath, const Pathname & newpath );
```

Comment:
`safe_rename` is just an internal helper and should not be exposed in the header.  The comment of `rename` could be updated to mention the fallback to `mv`.